ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/PutJobManagement_Test.java

```diff
@@ -0,0 +1,337 @@
+
+            assertThat(responseAfterModify.getMasterObjectListResult().getStartDate(), is(newDate));
+
+        } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    @Test
+    public void cancelJob() throws IOException, SignatureException, XmlProcessingException {
+        final String bucketName = "test_cancel_job";
+        try {
+            client.putBucket(new PutBucketRequest(bucketName));
+
+            final List<Ds3Object> objectsOne = new ArrayList<>();
+            final Ds3Object obj = new Ds3Object("testOne", 2);
+            objectsOne.add(obj);
+
+            final Ds3ClientHelpers.Job job =
+                    wrap(client).startWriteJob(bucketName, objectsOne);
+
+            final CancelJobSpectraS3Response response = client
+                    .cancelJobSpectraS3(new CancelJobSpectraS3Request(job.getJobId()));
+            response.checkStatusCode(204);
+
+            assertTrue(client.getActiveJobsSpectraS3(new GetActiveJobsSpectraS3Request())
+                    .getActiveJobListResult().getActiveJobs().isEmpty());
+
+        } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    @Test
+    public void truncateJobCancelWithOutForce() throws IOException, SignatureException, XmlProcessingException, URISyntaxException, InterruptedException {
+
+        final int testTimeOutSeconds = 5;
+        final String bucketName = "test_truncate_cancel_job";
+        final String book1 = "beowulf.txt";
+        final Path objPath1 = ResourceUtils.loadFileResource(RESOURCE_BASE_NAME + book1);
+        final Ds3Object obj1 = new Ds3Object(book1, Files.size(objPath1));
+        final Ds3Object obj2 = new Ds3Object("place_holder", 5000000);
+        final Path dirPath = FileSystems.getDefault().getPath("output");
+
+        try {
+            client.putBucket(new PutBucketRequest(bucketName));
+            Ds3ClientHelpers.wrap(client).ensureBucketExists(bucketName);
+
+            final Ds3ClientHelpers.Job putJob = Ds3ClientHelpers.wrap(client).startWriteJob(bucketName, Lists.newArrayList(obj1, obj2));
+            final UUID jobId = putJob.getJobId();
+            final SeekableByteChannel book1Channel = new ResourceObjectPutter(RESOURCE_BASE_NAME).buildChannel(book1);
+            client.putObject(new PutObjectRequest(bucketName, book1, book1Channel, jobId, 0, Files.size(objPath1)));
+
+            //make sure black pearl has updated it's job to show 1 object in cache
+            final long startTime = System.nanoTime();
+            long cachedSize =0;
+            while (cachedSize == 0) {
+                Thread.sleep(500);
+                final MasterObjectList mol = client.getJobSpectraS3(new GetJobSpectraS3Request(jobId)).getMasterObjectListResult();
+                cachedSize = mol.getCachedSizeInBytes();
+                assertTrue((System.nanoTime() - startTime)/1000000000 < testTimeOutSeconds );
```

Comment:
same comment about using the new test dependency.  You also may want to write a method method to do the `(System.nanoTime() - startTime)/1000000000` into a private static method at the bottom, since this is the 3rd or 4th time this exact code has been written.